spring-cloud-kubernetes-client-autoconfig/src/main/java/org/springframework/cloud/kubernetes/client/KubernetesClientAutoConfiguration.java

```diff
@@ -46,9 +47,11 @@
 
 	@Bean
 	@ConditionalOnMissingBean
-	public ApiClient apiClient() throws IOException {
```

Comment:
I would rather get this value from `KubernetesClientProperties`.  The default should be `application`.  

Why is this user agent needed?

Reply:
> I would rather get this value from KubernetesClientProperties. The default should be application.

updated

> Why is this user agent needed?

most kubernetes admins are tracking the resource requests by [auditing](https://kubernetes.io/docs/tasks/debug-application-cluster/audit/). and the user-agent header is one of the most useful things that helps us to understanding the source of the requests. otherwise the default user-agent will be [hard-coded](https://github.com/kubernetes-client/java/blob/master/kubernetes/src/main/java/io/kubernetes/client/openapi/ApiClient.java#L106)